src/plugins/dataflow/components/dataflow-program.tsx

```diff
@@ -371,6 +372,18 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
         }
         break;
     }
+    this.updateSharedProgramData();
```

Comment:
Because the sharedProgramData is volatile, we'll have to run this also in the snapshot handler. This way a readOnly view will have updated data when the snapshot changes.